### DISCUSSION TITLE
New install: hooking up RS-485 to pump

### DISCUSSION THREAD

**Discussion by chadhobson** [tagyoureit/nodejs-poolController#792] [Nixie]
I hate to keep asking questions but I feel like the wiki only takes you 95% of the way on everything I've done so far and then I need my hand held for the last 5%, so my apologies.

I've taken the enclosure out to the pool pad and plugged it into my Intelliflo i1 pump's RS-485 port. The other end of that wire is connected to my MEGA-BAS RS-485 port. Both wires are in the correct spot.

Pi has serial port enabled and I see it.

REM is set to RS-485 pass thru on the MEGA-BAS. This is where the wiki ends and now I start trying random stuff to get it working.

Go back to njsPC dash panel and into settings. Add new pump. No clue what my Intelliflo i1 is from the list (V/VF/VSF), so I try all three and.... nothing.

Dash doesn't show pump's GPM or watts and I don't have a clue how to debug it.

Help please? Again?

Thanks!

**Comment by rstrouse** [comment]
No worries that is what this forum is for. 

This should be set up as an IntelliFlo VS. You should be receiving messages that you can see these in https://github.com/rstrouse/nodejs-poolController-dashPanel/wiki/Message-Manager. Please make sure you have configured it on the drive as if it is being controlled by a Pentair OCP.

**Comment by rstrouse** [comment]
If the pump address, configuration on the pump, and wiring is correct the pump should say Display not Active.

**Reply by chadhobson** [reply]
The pump address is 1 on REM. Its also set for 1 on the pump itself. The pump wiring is green to A and yellow to B on the leftmost MEGA-BAS RS-485 port. Display on the pump is unchanged from what I normally see and I have full control over it. No messages coming in to Message Manager. Doesn't appear to be working at all, so I've missed something in the beginning.

I suspect this is a typo, but what does "drive" mean in this context?

"Please make sure you have configured it on the drive as if it is being controlled by a Pentair OCP."

**Reply by rstrouse** [reply]
Lets see what is being sent to the pump.  Capture a replay.  No need to capture a configuration reload so uncheck that box if it is checked.


https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution

**Reply by chadhobson** [reply]
Here's the zip file, if thats OK?

[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/11776353/replay.zip)

**Reply by rstrouse** [reply]
That was perfect.  You do not have any circuits assigned to your pump.  Add the pool circuit and assign it a speed.  Then turn on that circuit and then perform the replay again.

**Reply by rstrouse** [reply]
Also please turn off the MQTT interface that you have enabled for Home Assistant.  This is not required or desired and you do not have it set up correctly so it is constantly trying to authenticate with the broker.  You should use the @crewski integration which does not use MQTT and connects directly to the njsPC sockets.

**Reply by chadhobson** [reply]
[Latest Replay](https://github.com/tagyoureit/nodejs-poolController/files/11777677/replay.zip)

And MQTT is off now.

**Reply by rstrouse** [reply]
Please go into the gears menu then select the controller => Interfaces tab.  Then click on the Home Assistant interface and uncheck the Enabled checkbox.  Finally click Save Interface.


Next click on the Comma tab and open the primary port.  Change the port name to /dev/ttyS0.  Check the enabled checkbox and save the port.

**Reply by chadhobson** [reply]
Getting closer after those changes. I now see message activity in the Comms tab but its all failures. Nothing in MessageManager.

Here's a new [replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/11779093/replay.zip) that shows the errors. Thanks again for the help!

**Reply by rstrouse** [reply]
Alright we are now sending out messages to our pump but it is not responding.  If you are not seeing these outbound messages in messageManager then either the pump messages are not selected in the filter or the start/stop log isn't activated.

![image](https://github.com/tagyoureit/nodejs-poolController/assets/47839015/3d8c6b44-2ff3-4a63-84dd-0e5dadfa5f60)


If this isn't an issue with wiring then it has to be with the configuration of the pump drive (the drive is the pancake hat that sits on top of the motor and tells it how fast or slow to go) or with the configuration of the BAS.

If you have not updated the firmware for the MEGA-BAS please do this now.  But whatever you do shut down REM before trying to update it.  It will drive you nuts if you don't because it will look like it succeeded but the firmware will not actually install.

If you are running pm2 type `pm2 stop all` before starting the update procedure.
https://github.com/SequentMicrosystems/megabas-rpi/blob/master/update/README.md

If you have already done this or just completed it then type the following commands to disable the modbus functions of the MEGA-BAS
`cd ~/megabas-rpi`
`megabas 0 rs485wr 0 9600 1 0 1`
`pm2 restart all`

If this doesn't get the pump to respond, lets revisit the wiring.  I see collisions in your logs indicating that something it trying to respond to our request but it is gibberish.

**Reply by chadhobson** [reply]
I have completed all these steps. Some output from the MEGA-BAS...

```
chadhobson@nixie:~$ megabas 0 board
Firmware ver 01.09, CPU temperature 36 C, Power source 21.24 V, Raspberry 5.13 V
chadhobson@nixie:~$ megabas 0 rs485rd
<mode> <baudrate> <stopbits> <parity> <add> 0 38400 1 0 1
```

And ticking that filter seems to have brought messageManager to life now but still only one-way comms to the pump...

<img width="1289" alt="Screenshot 2023-06-17 at 6 09 47 PM" src="https://github.com/tagyoureit/nodejs-poolController/assets/1839161/596b63f4-3552-4c33-b8f1-a624a628c645">

As far as wiring goes, I don't know what to say. Its a green and yellow wire right off the pump (this is the only thing connected other than a temp probe) and into the MEGA-BAS port. Its pretty much impossible to screw up, IMO.

Thoughts?

**Comment by chadhobson** [comment]
Here's some output from fresh restarts for the controller, the web panel, and REM.

**Controller**
```
[6/16/2023, 4:58:49 PM] info: The current git branch output is master
[6/16/2023, 4:58:49 PM] info: The current git commit output is b61243140858f4276f583fb3b873e9e3d30866cf
[6/16/2023, 4:58:49 PM] info: Starting up SSDP server
[6/16/2023, 4:58:49 PM] info: Init mqtt interface: Home Assistant
[6/16/2023, 4:58:49 PM] info: Initializing MQTT client Home Assistant
[6/16/2023, 4:58:49 PM] info: Init rem interface: Relay Equipment Manager
[6/16/2023, 4:58:49 PM] info: Opening Relay Equipment Manager socket on http://127.0.0.1:8080
[6/16/2023, 4:58:49 PM] info: Checking njsPC versions...
[6/16/2023, 4:58:49 PM] info: Starting Pool System nixie
[6/16/2023, 4:58:49 PM] info: Initializing Nixie Control Panel for Nixie Single Body
[6/16/2023, 4:58:49 PM] info: Initializing Intake/Return valves
[6/16/2023, 4:58:49 PM] info: Initializing Nixie Controller
[6/16/2023, 4:58:49 PM] info: Initializing Nixie body Pool
[6/16/2023, 4:58:49 PM] info: Initializing Filter Filter
[6/16/2023, 4:58:49 PM] info: Auto-backup initialized Last Backup: 1969-12-31T19:00:00.000-0500
[6/16/2023, 4:58:49 PM] info: Initializing Nixie circuit Pool
[6/16/2023, 4:58:49 PM] info: Initializing Nixie Pump 50-Pump 1
[6/16/2023, 4:58:49 PM] info: Nixie Controller Initialized
[6/16/2023, 4:58:49 PM] info: Nixie Single Body control board initialized
[6/16/2023, 4:58:49 PM] info: Server is now listening on 0.0.0.0:4200 - 192.168.1.251:4200
```

**Web**
```
Updated configuration file
info: Server is now listening on 0.0.0.0:5150 - 192.168.1.251:5150

```
**REM**
```
info: Starting up SSDP server
info: Initializing GPIO Pins 0
info: Initializing SPI Bus #0
info: SPI Bus #0 Initialized
info: Initializing SPI Bus #1
info: SPI Bus #1 Initialized
info: Initializing i2c Interface
info: Detecting i2c Buses
info: Detecting i2c Buses /sys/class/i2c-dev/i2c-1
info: Initializing 1-Wire Interface
info: Detecting 1-Wire Buses
info: Initializing Generic Devices
info: Initializing i2c Bus #1
error: 1-Wire interface not loaded on os.  Please enable through OS config. Error: Command failed: lsmod | grep w1_gpio
info: Device 10k Temperature Probe initialized for generic device 1 - Air Temp Probe
info: Generic Devices Initialized
info: Server is now listening on 0.0.0.0:8080
info: Scanning i2c Bus #1
info: Connected to njsPC - automatic : http://localhost:4200
[ 39, 72 ]
info: Found I2C device Multi-Relay Board at address: 39 - (0x27)
info: Found I2C device Sequent MEGA-BAS at address: 72 - (0x48)
info: Device Multi-Relay Board initialized for i2c bus #1 address 39
info: Device Sequent MEGA-BAS initialized for i2c bus #1 address 72
info: i2c Bus #1 Initialized
info: New socket client connected JoCiP-lmIe-tLRmsAAAB -- 127.0.0.1
info: New socket client connected RKOg4pFkoUf-c6YfAAAD -- 192.168.1.111
info: [5:01:44 PM] 192.168.1.111 GET /devices/state?null {}
```

**Comment by rstrouse** [comment]
Flip the wires.

**Reply by chadhobson** [reply]
Boom! Now its working! Thanks again for the assist. I'm going to play with this setup a bit and may have other questions but the pump is showing up now so all good!

**Reply by rstrouse** [reply]
Glad it is working.  It makes much more sense when the pump is actually communicating.  Find the lowest rpm that you want the pump to run at for the pool circuit,  Typically, this is a couple hundred rpm more than what your chlorinator requires if you have one.  Then think of how much more power you would like for things like skimming, solar, or running your cleaner.  Finally, if this also runs pool features such as waterfalls, jets, or the like add those features to the mix for the pump circuits.

njsPC will look at the running options on the pool and figure out how fast the pump should run to get it all done.
